src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.cs

```diff
@@ -128,11 +139,20 @@ public AdjustmentRule[] GetAdjustmentRules()
             // as the rules now is public, we should fill it properly so the caller doesn't have to know how we use it internally
             // and can use it as it is used in Windows
 
-            AdjustmentRule[] rules = new AdjustmentRule[_adjustmentRules.Length];
+            List<AdjustmentRule> rulesList = new List<AdjustmentRule>();
```

Comment:
Is it likely that we'll end up with _adjustmentRules.Length items, or close to it?  If yes, might want to pass _adjustmentRules.Length into the list.  (If it's super likely we'll end up with that exact number, you could also just stick with an array here, track how many you've stored, and return the array if it's filled or else resize if it's not.  But I assume this code isn't that perf sensitive.)

Reply:
We'll end up with more than the array length in more of the cases. I can start the list with that length and let it grow as needed. This is not perf sensitive as I expect users will call it once and cache the result if they need to.